tools/fog-local-network/fog_conformance_tests.py

```diff
@@ -764,6 +765,54 @@ def run(self):
             self.follow_up_balance_check(wallet3, {8: 1}, [8])
             self.follow_up_balance_check(wallet4, {8: 14}, [8])
 
+
+        #######################################################################
+        # Test that when we change the expiry window, we record when was the 
+        # last time the report has been cached.  The desired behavior is that
+        # when the pubkey expiry window is N blocks, then the last
+        # reported published block should be N blocks ago.
+        #######################################################################
+
+        # Expecting index=7
+        index1 = self.fog_ingest.get_last_report_published_block_index()
+        print("===> last report published block index: ", index1)
+
+        # Add nothing
+        # Take 4 from 4
+        credits7x = [{'account': 0, 'amount': 0},
+                     {'account': 1, 'amount': 0},
+                     {'account': 2, 'amount': 0},
+                     {'account': 3, 'amount': 0},
+                     {'account': 4, 'amount': 0}]
+        key_images7x = [block7_key_images[0]]
+        print("===> Key images spent in Block 7x: ", key_images7x)
+        block7x_key_images = ledger1.add_block(credits7x, key_images7x, fog_pubkey)
+        ledger2.add_block(credits7x, key_images7x, fog_pubkey)
+        print("===> Key images for new transactions in Block 7x: ", block7x_key_images)
+        time.sleep(1)
+
+        # Expecting index=8
+        index2 = self.fog_ingest.get_last_report_published_block_index()
+        print("===> last report published block index: ", index2)
+
+        assert (index2 != index1) and (index2 > index1)
+
+        # Add 4 back to 4
+        # Take nothing
+        credits7y = [{'account': 4, 'amount': 4}]
+        key_images7y = [block7x_key_images[0]]
+        print("===> Key images spent in Block 7y: ", key_images7y)
+        block7y_key_images = ledger1.add_block(credits7y, key_images7y, fog_pubkey)
+        ledger2.add_block(credits7y, key_images7y, fog_pubkey)
+        print("===> Key images for new transactions in Block 7y: ", block7y_key_images)
+        time.sleep(3)
+
+        # Expecting index=9 (works when sleep>1, reported index=8 when sleep=1)
+        index3 = self.fog_ingest.get_last_report_published_block_index()
+        print("===> last report published block index: ", index3)
+
+        assert (index3 != index2) and (index3 > index2)
```

Comment:
I think the assert should check for exact values since in this test we know exactly at which block we expect to be.